packages/adblocker/src/engine/engine.ts

```diff
@@ -1503,17 +1575,30 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
       // If there was a redirect match and no exception was found, then we
       // proceed and process the redirect rule. This means two things:
       //
-      // 1. Check if a redirect=none rule was found, which acts as exception.
-      // 2. If no exception was found, prepare `result.redirect` response.
+      // 1. Check if there's a removeparam exception was found, which acts as exception.
+      // 2. Check if a redirect=none rule was found, which acts as exception.
+      // 3. If no exception was found, prepare `result.redirect` response.
       if (
         result.filter !== undefined &&
         result.exception === undefined &&
-        result.filter.isRedirect()
+        result.filter.isRedirectable()
       ) {
-        if (redirectNone !== undefined) {
-          result.exception = redirectNone;
+        if (result.filter.isRemoveParam()) {
+          if (removeparamExceptionFilter === undefined) {
+            result.redirect = {
+              body: '',
+              contentType: 'text/plain',
+              dataUrl: redirectUrl!.toString(),
+            };
```

Comment:
It seems a bit counter-intuitive to me to reuse the dataUrl in order to express the removeparam result. Why not return a list of URL modifiers to be interpreted by the client code and adapted based on available capability of the platform? (iOS, Manifest v2, Manifest v3, might not have the best way to modify request parameters) Ideally the adblocker library should not need to think about these considerations and return a more "abstract" kind of response.

Reply:
Fixed by 76e6a5128c1bcb61bced742e2a2868f38bc19c9d